sphinxlint/__main__.py

```diff
@@ -118,13 +123,35 @@ def _check_file(todo):
     return check_file(*todo)
 
 
-def print_results(results):
-    """Print results (or a message if nothing is to be printed)."""
+def sort_errors(results, sorted_by):
+    """Flattens and potentially sorts errors based on user prefernces"""
+    if not sorted_by:
+        for results in results:
+            yield from results
+        return
+    errors = list(error for errors in results for error in errors)
+    # sorting is stable in python, so we can sort in reverse order to get the
+    # ordering specified by the user
+    for field in reversed(sorted_by):
+        if field == 'error_type':
+            errors.sort(key=lambda error: error.checker_name)
+        elif field == "filename":
+            errors.sort(key=lambda error: error.filename)
+        elif field == "line":
+            errors.sort(key=lambda error: error.line_no)
+        else:
+            raise ValueError(f"Unsupported sorting field {field}")
```

Comment:
I won't be happy to have sphinx-lint run multiple seconds before telling me that I have a typo in my `--sort-by` argument.

Could we have this test done in `parse_args` instead?

Reply:
yes, I thought about it but was too lazy to do it properly -- I'll do this properly now.

Reply:
I can relate ;)